src/containers/BookForm.js

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable class-methods-use-this */
+/* eslint-disable react/no-unescaped-entities */
+// The form should contain a
+// book title input,
+// a select box with categories and a
+// submit button.
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import './BookForm.css';
+import { createBook } from '../actions';
+import randomId from '../helpers/randomId';
+
+// eslint-disable-next-line no-unused-vars
+class BookForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      completed: true,
+      category: '',
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const { target } = event;
+    const { dispatch, books } = this.props;
+
+    const book = {
+      id: randomId(),
+      title: target.title.value,
+      category: target.category.value,
+    };
+
+    createBook(book, dispatch);
+    const bookInStore = books.filter(b => b.id === book.id);
+    if (bookInStore !== book) {
+      // eslint-disable-next-line no-console
+      console.error('Data inconsistency handling book submit. Book in store is not the same as the book submited.');
```

Comment:
I use console.error() because I don't know a better way to show the user that the app has data inconsistency.

Reply:
Thanks, Sergio Torres for the review!